include/os/macos.h

```diff
@@ -5,6 +5,17 @@
 
 #include <libkern/OSByteOrder.h>
 #include <mach/vm_statistics.h>
+#include <pthread.h>
 #define bswap_32(x) OSSwapInt32(x)
 #define bswap_64(x) OSSwapInt64(x)
 #define ENVIRON NULL
+
+/*
+ * On pthread_exit, it expects to be dealing with libmalloc
+ * rather than isoalloc at releasing time
+ */
+__attribute__((weak)) void _pthread_tsd_cleanup(pthread_t);
+void _pthread_tsd_cleanup(pthread_t p)
```

Comment:
Should this be wrapped in `#ifdef __APPLE__` ?

Reply:
seems to be here https://github.com/struct/isoalloc/blob/cd6a1c1f38b3a673b53befbb7d41a74aa8e72d6c/include/iso_alloc_internal.h#L20

Reply:
Ah I misread where this was located. You're correct.